### PR TITLE
update the get subscription v2 endpoint to return subscription details

### PR DESCRIPTION
**Pull request by tindNan**

### CODE REVIEW

**Comment by tindNan** [comment]
@UwaoEvan you can double check if this is okay now

File: src/routes/subscriptions/v2/index.ts [hunk 1]
```diff
@@ -1,6 +1,12 @@
-import { and, desc, eq } from "drizzle-orm";
+import { and, desc, eq, isNotNull } from "drizzle-orm";
```
Comment (UwaoEvan) [inline src/routes/subscriptions/v2/index.ts:1]: remove this unused import.

File: src/routes/subscriptions/v2/index.ts [hunk 2]
```diff
@@ -17,20 +23,67 @@ const subscriptionsRouters: FastifyPluginAsync = async (
...
+    if (!currentUser) {
+      throw fastify.httpErrors.notFound("Could not find the user");
+    }
 
-    // TODO: align on format to check if user has a valid subscription, throw a 404 error or just return an empty object
-    // for now it will return an empty object
+    if (currentUser.user.clientId) {
+      const currentGroupPlan = await fastify.db.query.groupPlan.findFirst({
+        where: eq(groupPlan.clientId, currentUser.user.clientId),
+      });
```
Comment (UwaoEvan) [inline src/routes/subscriptions/v2/index.ts:39]: I think we should just get the group plan which hasn't expired. We can add more filters to this query.
groupPlan.expireTime > now

File: src/routes/subscriptions/v2/index.ts [hunk 3]
```diff
@@ -17,20 +23,67 @@ const subscriptionsRouters: FastifyPluginAsync = async (
...
+
+      if (currentGroupPlan && currentGroupPlan.expireTime < new Date()) {
+        return {
+          subscription_name: undefined,
+          expiry_date: currentGroupPlan.expireTime,
+          is_group_plan: false,
+          group_plan: currentUser.client.companyName,
+        };
+      }
+    }
```
Comment (UwaoEvan) [inline src/routes/subscriptions/v2/index.ts:58]: You know the issue with this  condition here is that lets say the user has a group plan which is expired, this user won't be able to see his/her new B2C subscription. The API will be returning the expired group plan everytime.

File: src/routes/subscriptions/v2/index.ts (hunk 2 above)
Comment (tindNan) [reply src/routes/subscriptions/v2/index.ts:39]: +1

File: src/routes/subscriptions/v2/index.ts (hunk 3 above)
Comment (tindNan) [reply src/routes/subscriptions/v2/index.ts:58]: noice

Comment (tindNan) [reply src/routes/subscriptions/v2/index.ts:58]: so if the group plan is expired, then we just go ahead and look up items in the subscriptions page

Comment (tindNan) [reply src/routes/subscriptions/v2/index.ts:52]: made some updates for your review

**Review by UwaoEvan — APPROVED** [review]

**Pull request merged**